### PR TITLE
Be able to specify user_id when creating keypair

### PR DESCRIPTION
**Pull request by smutel**
Implement #1230

### CODE REVIEW

**Comment by smutel** [comment]
Currently, I didn't implement the data part. I would like to have a review from someone first.

**Comment by smutel** [comment]
Could you please start the CI again ?

**Comment by ozerovandrei** [comment]
recheck

**Comment by smutel** [comment]
Could you please confirm that the CI is crashing due to my code ?

**Comment by ozerovandrei** [comment]
@smutel yeah, import test is failing:

```
2022-02-17 19:11:30.266925 | ubuntu-focal |     TestAccComputeV2Keypair_importBasic: import_openstack_compute_keypair_v2_test.go:12: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected
.
2022-02-17 19:11:30.267095 | ubuntu-focal |
2022-02-17 19:11:30.267233 | ubuntu-focal |         (map[string]string) {
2022-02-17 19:11:30.267297 | ubuntu-focal |         }
2022-02-17 19:11:30.268166 | ubuntu-focal |
2022-02-17 19:11:30.268284 | ubuntu-focal |
2022-02-17 19:11:30.268414 | ubuntu-focal |         (map[string]string) (len=1) {
2022-02-17 19:11:30.268561 | ubuntu-focal |          (string) (len=7) "user_id": (string) ""
2022-02-17 19:11:30.268611 | ubuntu-focal |         }
```

File: openstack/resource_openstack_compute_keypair_v2.go [hunk 1]
```diff
@@ -98,7 +111,19 @@ func resourceComputeKeypairV2Read(_ context.Context, d *schema.ResourceData, met
 		return diag.Errorf("Error creating OpenStack compute client: %s", err)
 	}
 
-	kp, err := keypairs.Get(computeClient, d.Id(), keypairs.GetOpts{}).Extract()
+	userID := d.Get("user_id").(string)
+	if userID != "" {
+		computeClient.Microversion = computeKeyPairV2UserIDMicroversion
+	}
+
+	log.Printf("[DEBUG] User ID %s", userID)
```
Comment (ozerovandrei) [inline openstack/resource_openstack_compute_keypair_v2.go:119]: I don't think that those debug messages are needed.
Also printing user ID seems insecure.

File: openstack/resource_openstack_compute_keypair_v2.go [hunk 2]
```diff
@@ -109,6 +134,9 @@ func resourceComputeKeypairV2Read(_ context.Context, d *schema.ResourceData, met
 	d.Set("public_key", kp.PublicKey)
 	d.Set("fingerprint", kp.Fingerprint)
 	d.Set("region", GetRegion(d, config))
+	if userID != "" {
```
Comment (ozerovandrei) [inline openstack/resource_openstack_compute_keypair_v2.go:137]: Can't we just read it from `kp` struct? Is it not populated in HTTP response?
Either way I guess we need to set `Computed: true` on the new resource attribute if we populate it in the `resourceComputeKeypairV2Read` function.

File: openstack/resource_openstack_compute_keypair_v2.go (hunk 1 above)
Comment (smutel) [reply openstack/resource_openstack_compute_keypair_v2.go]: The user_id will be in the tfstate also and I am not sure that it's a confidential item.
Anyway I removed this comment.

File: openstack/resource_openstack_compute_keypair_v2.go (hunk 2 above)
Comment (smutel) [reply openstack/resource_openstack_compute_keypair_v2.go:137]: I set `Computed: true`

**Comment by smutel** [comment]
Any help to pass the CI will be helpful ...

**Comment by ozerovandrei** [comment]
recheck

**Comment by ozerovandrei** [comment]
Error is unrelated:
```
TestAccSFSV2Share_update: resource_openstack_sharedfilesystem_share_v2_test.go:71: Step 1/4 error: Check failed: Check 5/5 error: Metadata does not match. Expected map[] but got value
```

**Review by ozerovandrei — APPROVED** [review]
LGTM, thank you @smutel

**Comment by smutel** [comment]
Is-it possible to have a new release with this feature please ?

**Comment by smutel** [comment]
Hello @ozerovandrei,

Did you plan to create a new release soon ?

Thanks.

**Comment by ozerovandrei** [comment]
Hello @smutel 
It would be great to fix CI first to ensure that our provider works as expected prior releasing a new version: https://github.com/terraform-provider-openstack/terraform-provider-openstack/issues/1366
Unfortunately I haven't been able to find any time to do it yet.

**Pull request merged**